### PR TITLE
adopt curtin capabilities related to kernel in lowest layer

### PR DESCRIPTION
**Pull request by dbungert**

### CODE REVIEW

**Review by Chris-Peterson444 — APPROVED** [review]

File: subiquity/models/kernel.py [hunk 1]
```diff
@@ -40,11 +40,9 @@ def needed_kernel(self) -> Optional[str]:
...
-
-        return {
-            "kernel": {
-                "package": self.needed_kernel,
-            },
-        }
+            result["kernel"]["install"] = False
+        else:
+            result["kernel"]["package"] = self.needed_kernel
+        return result
```
Comment (mwhudson) [inline subiquity/models/kernel.py:48]: tangent: would be nice if we could use the rich curtin config types we've added here wouldn't it ...

File: subiquity/models/kernel.py [hunk 2]
```diff
@@ -40,11 +40,9 @@ def needed_kernel(self) -> Optional[str]:
         return self.metapkg_name
 
     def render(self):
+        result = {"kernel": {"remove_existing": True}}
         if self.curthooks_no_install:
```
Comment (mwhudson) [inline subiquity/models/kernel.py:44]: I don't think any core remains to set curthooks_no_install to True, so can this be tidied up?

**Review by mwhudson — COMMENTED** [review]
nice simplification, I think you can go a touch further though

File: subiquity/models/kernel.py (hunk 2 above)
Comment (dbungert) [reply subiquity/models/kernel.py:44]: I feel tentative on if it's going to be useful or not, so I'm dropping it but isolating that change to a distinct commit.

**Pull request merged**